### PR TITLE
Updated downsampler in wvp + improved error handling

### PR DESCRIPTION
**Pull request by mrrostam**
- updated the internal downsampler's wasm file to the latest version.
- improved error handling by adding `ds-failed` message and rejecting the promise in case of init errors
- moved demo and package files to different folders to avoid recursive dependencies while building the demo
- added github's action

### CODE REVIEW

File: demo/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "web-voice-processor-demo",
-  "version": "2.0.3",
+  "version": "2.0.0",
```
Comment (ErisMik) [inline demo/package.json:3]: We don't publish this anywhere right? So this won't be an issue?

File: package/src/worker_types.ts [hunk 2]
```diff
@@ -29,6 +29,11 @@ export type DownsamplingWorkerResponseReady = {
   command: 'ds-ready';
 };
 
+export type PorcupineWorkerResponseFailed = {
```
Comment (ErisMik) [inline package/src/worker_types.ts]: I'm guessing this should be `DownsamplingWorkerResponseFailed`?

File: demo/package.json (hunk 1 above)
Comment (mrrostam) [reply demo/package.json:3]: correct, to me having such a specific version for the demo just looked weird :)

File: package/src/worker_types.ts (hunk 2 above)
Comment (mrrostam) [reply package/src/worker_types.ts]: good catch!

**Review by ErisMik — APPROVED** [review]

**Pull request merged**